tests/test_performance_benchmarks.py

```
@@ -0,0 +1,415 @@
+    def test_performance_trend_monitoring(self):
+        """Test that we can detect performance trends."""
+        tracker = PerformanceTracker()
+        
+        # Simulate measurements over time with degrading performance
+        baseline_time = 0.01
+        measurements = []
+        
+        for i in range(5):
+            # Simulate gradually slower operation
+            def slow_operation():
+                time.sleep(baseline_time * (1 + i * 0.1))
```

Ruff (0.12.2)

Function definition does not bind loop variable `i`

(B023)